lib/gears/timer.lua

```diff
@@ -215,14 +219,18 @@ function timer.start_new(timeout, callback)
     return t
 end
 
---- Create a timeout for calling some callback function.
--- This function is almost identical to `gears.timer.start_new`. The only difference
--- is that this does not prevent the callback function from being garbage
--- collected. After the callback function was collected, the timer returned
--- will automatically be stopped.
+--- Create a simple timer for calling the `callback` function continuously.
+--
+-- This function is almost identical to `gears.timer.start_new`. The only
+-- difference is that this does not prevent the callback function from being
+-- garbage collected.
+-- In addition to the conditions in `gears.timer.start_new`,
+-- this timer will also stop if `callback` was garbage collected since the
+-- previous run.
```

Comment:
This new wording suggests that the timer will stop immediately after the garbage collector run. It suggests that this (theoretical) example should work:

```lua
print(timer.started) -- true
-- drop the last reference to the callback function here
collectgargabe("collect") -- callback function is collected
print(timer.started) -- false
```

Whereas in reality, if the timer only runs every 5 minutes and the callback function is collected right after the timer triggered, it will take another 5 minutes before the timer reports `timer.started == false`.

Reply:
What about:
```suggestion
-- In addition to the conditions in `gears.timer.start_new`,
-- this timer will also stop at its next run if `callback` was garbage collected.
```

Reply:
Hm... actually, it doesn't change anything. Let's keep your version!